### ISSUE TITLE
Request: An alternative approach to ability modifiers derived from light/medium/heavy armor

### ISSUE THREAD

**Issue by krbz999** [foundryvtt/dnd5e#2260]
Instead of heavy armor being hard-coded to never add Dexterity, it would be nice if they simply had "0" meaning "don't add an ability modifier, ever". Currently it is not possible to create, for example, a magic set of heavy armor that *does* add Dexterity.

Additionally, it would be nice if each armor could select what ability was used, default being Dex. On the actor, rather than `attributes.ac.dex`, this could be split into `attributes.ac.ability` and `attributes.ac.mod`, both derived values (the ability modifier added on top of AC, from the armor, and the total modifier capped by what is set in the armor's details).

**Comment by Fyorl** [comment]
I don't think we can use `0` for this. There would be no way to express 'add your dexterity modifier if it is negative, but not positive'. Kind of like how two-weapon fighting works with regards to adding your ability modifier to the off-hand damage.

**Comment by Ikaguia** [comment]
> I don't think we can use `0` for this. There would be no way to express 'add your dexterity modifier if it is negative, but not positive'. Kind of like how two-weapon fighting works with regards to adding your ability modifier to the off-hand damage.

Is there a use case for adding the modifier only if it's negative? You either add it up to a certain value, or not at all. 0 would mean not at all in this case.

**Comment by Fyorl** [comment]
> Is there a use case for adding the modifier only if it's negative?

The issue was raised to support this use case:

> Currently it is not possible to create, for example, a magic set of heavy armor that *does* add Dexterity.

I'm saying that by using `0` for this, you close the door on the use case of having armour that adds up to 0 of your dex mod (i.e. how it currently works).

Both are hypothetical use cases that have no examples in official content, so why is one more valid than the other?
